ckanext/validation/jobs.py

```diff
@@ -54,54 +55,62 @@ def run_validation_job(resource):
-            if dataset[u'private'] and pass_auth_header:
+                t.config.get('ckanext.validation.pass_auth_header', True))
+            if dataset['private'] and pass_auth_header:
                 s = requests.Session()
                 s.headers.update({
-                    u'Authorization': t.config.get(
-                        u'ckanext.validation.pass_auth_header_value',
+                    'Authorization': t.config.get(
+                        'ckanext.validation.pass_auth_header_value',
                         _get_site_user_api_key())
                 })
 
-                options[u'http_session'] = s
+                options['http_session'] = s
 
     if not source:
-        source = resource[u'url']
-
-    schema = resource.get(u'schema')
-    if schema and isinstance(schema, str):
-        if schema.startswith('http'):
-            r = requests.get(schema)
-            schema = r.json()
+        source = resource['url']
+
+    schema = resource.get('schema')
+    if schema:
+        if isinstance(schema, str):
+            if schema.startswith('http'):
+                r = requests.get(schema)
+                schema = r.json()
         else:
-            schema = json.loads(schema)
+            schema = json.dumps(schema)
 
-    _format = resource[u'format'].lower()
+    _format = resource['format'].lower()
 
     report = _validate_table(source, _format=_format, schema=schema, **options)
 
     # Hide uploaded files
-    for table in report.get('tables', []):
-        if table['source'].startswith('/'):
-            table['source'] = resource['url']
-    for index, warning in enumerate(report.get('warnings', [])):
-        report['warnings'][index] = re.sub(r'Table ".*"', 'Table', warning)
-
-    if report['table-count'] > 0:
-        validation.status = u'success' if report[u'valid'] else u'failure'
-        validation.report = report
+    if type(report) == Report:
+        report = report.to_dict()
+    if 'tables' in report:
+        for table in report['tables']:
+            if table['source'].startswith('/'):
+                table['source'] = resource['url']
+    if 'warnings' in report:
+        for index, warning in enumerate(report['warnings']):
+            report['warnings'][index] = re.sub(r'Table ".*"', 'Table', warning)
+    if 'valid' in report and report['valid']:
+        validation.status = 'success' if report['valid'] else 'failure'
+        validation.report = json.dumps(report)
     else:
-        validation.status = u'error'
-        validation.error = {
-            'message': '\n'.join(report['warnings']) or u'No tables found'}
+        validation.status = 'error'
+        validation.report = json.dumps(report)
+        if 'tables' in report: 
+            validation.error = {
+                'message': [str(err) for err in report['tables'][0]['errors']] if len(report['tables'][0]['errors']) > 0 else 'No tables found'}
+        else:
+            validation.error = {'message': []}
```

Comment:
```suggestion
            validation.error = {'message': ['Error running validation']}
```

Is there another key in the report that can give a more meaningful message?